source/ftrack_connect_pipeline_3dsmax/host/engine/asset_manager.py

```diff
@@ -1,65 +1,73 @@
 # :coding: utf-8
-# :copyright: Copyright (c) 2014-2020 ftrack
+# :copyright: Copyright (c) 2014-2022 ftrack
 
 import time
-from ftrack_connect_pipeline import constants
+
+from ftrack_connect_pipeline import constants as core_constants
 from ftrack_connect_pipeline.host.engine import AssetManagerEngine
-from ftrack_connect_pipeline_3dsmax.asset import FtrackAssetNode
 from ftrack_connect_pipeline.asset.asset_info import FtrackAssetInfo
 from ftrack_connect_pipeline_3dsmax.utils import custom_commands as max_utils
 from ftrack_connect_pipeline_3dsmax.constants import asset as asset_const
+from ftrack_connect_pipeline_3dsmax.constants.asset import modes as modes_const
+from ftrack_connect_pipeline_3dsmax.asset import MaxFtrackObjectManager
+from ftrack_connect_pipeline_3dsmax.asset.dcc_object import MaxDccObject
 
 
 class MaxAssetManagerEngine(AssetManagerEngine):
-    ftrack_asset_class = FtrackAssetNode
     engine_type = 'asset_manager'
 
-    def __init__(self, event_manager, host_types, host_id, asset_type_name=None):
+    FtrackObjectManager = MaxFtrackObjectManager
+    '''FtrackObjectManager class to use'''
+    DccObject = MaxDccObject
+    '''DccObject class to use'''
+
+    def __init__(
+        self, event_manager, host_types, host_id, asset_type_name=None
+    ):
         '''Initialise AssetManagerEngine with *event_manager*, *host*, *hostid*
         and *asset_type_name*'''
         super(MaxAssetManagerEngine, self).__init__(
             event_manager, host_types, host_id, asset_type_name=asset_type_name
         )
 
+    @max_utils.run_in_main_thread
     def discover_assets(self, assets=None, options=None, plugin=None):
         '''
         Discover all the assets in the scene:
         Returns status and result
         '''
         start_time = time.time()
-        status = constants.UNKNOWN_STATUS
+        status = core_constants.UNKNOWN_STATUS
         result = []
         message = None
 
         result_data = {
             'plugin_name': None,
-            'plugin_type': constants.PLUGIN_AM_ACTION_TYPE,
+            'plugin_type': core_constants.PLUGIN_AM_ACTION_TYPE,
             'method': 'discover_assets',
             'status': status,
             'result': result,
             'execution_time': 0,
-            'message': message
+            'message': message,
         }
 
-        ftrack_asset_nodes = max_utils.get_ftrack_helpers()
+        ftrack_asset_nodes = max_utils.get_ftrack_nodes()
```

Comment:
Remember to update the utils then. In 3dsmax there is no ftrack node concept, those are helpers.